plugins/module_utils/network/nxos/utils/utils.py

```diff
@@ -189,7 +189,7 @@ class Version:
     """Simple class to compare arbitrary versions"""
 
     def __init__(self, version_string):
-        self.components = version_string.split(".")
+        self.components = list(map(int, version_string.split(".")))
```

Comment:
When each part of the version was of type str, it does not work correctly once the version exceeds 10. So We need to convert to str to int.

```
>>> ["10", "3"] > ["9", "1"]
False
>>> "10" > "9"
False
>>>
```